tests/Symbolica.Extensions.Configuration.FSharp.Tests/BindTests.fs

```diff
@@ -144,57 +218,171 @@ module OptValue =
 
-        test <@ key |> Bind.optValue |> Binder.eval section = Success(None) @>
+        test
+            <@ Bind.optValue key Bind.string
+               |> Binder.eval section = Success(None) @>
 
-module OptValueOf =
-    [<Property(Arbitrary = [| typeof<Arb.NotNullString>
-                              typeof<ConfigurationArb> |])>]
-    let ``when prop exists and can be decoded should be Success value``
-        (ConfigurationPath path)
-        (ConfigurationKey key)
-        (x: int)
-        =
+    [<Property(Arbitrary = [| typeof<ConfigurationArb> |])>]
+    let ``when value cannot be decoded should be Failure`` (ConfigurationPath path) (ConfigurationKey key) =
         let section =
             { Children =
                 [ { Children = Seq.empty
                     Path = ConfigurationPath.Combine(path, key)
-                    Value = x |> string } ]
+                    Value = "string" } ]
               Path = path
               Value = null }
 
-        test <@ Bind.optValueOf Decode.int key |> Binder.eval section = Success(Some(x)) @>
+        test
+            <@ Bind.optValue key Bind.int |> Binder.eval section = Failure(
+                [ "Could not decode 'string' as type 'System.Int32'." ]
+            ) @>
 
-    [<Property(Arbitrary = [| typeof<Arb.NotNullString>
-                              typeof<ConfigurationArb> |])>]
-    let ``when prop does not exist should be Success None`` (ConfigurationPath path) (ConfigurationKey key) (x: int) =
+    [<Property(Arbitrary = [| typeof<ConfigurationArb> |])>]
+    let ``when section has children should be Failure`` (ConfigurationPath path) (ConfigurationKey key) =
         let section =
             { Children =
-                [ { Children = Seq.empty
+                [ { Children =
+                      [ { Children = Seq.empty
+                          Path = ConfigurationPath.Combine(path, key)
+                          Value = "Value" } ]
                     Path = ConfigurationPath.Combine(path, key)
-                    Value = x |> string } ]
+                    Value = "Value" } ]
               Path = path
               Value = null }
 
-        let missingKey = $"{key}-1"
+        test
+            <@ Bind.optValue key Bind.string
+               |> Binder.eval section = Failure(
+                [ $"Expected a simple value at '{ConfigurationPath.Combine(path, key)}' but found an object." ]
+            ) @>
+
+module Bool =
+    [<Property>]
+    let ``should be Success value if can be converted to bool`` value =
+        test <@ value |> string |> Binder.run Bind.bool = Success(value) @>
 
+    [<Fact>]
+    let ``should be Failure if string can not be converted to bool`` =
         test
-            <@ missingKey
-               |> Bind.optValueOf Decode.int
-               |> Binder.eval section = Success(None) @>
+            <@ "string" |> Binder.run Bind.bool = Failure([ $"Could not decode 'string' as type 'System.Boolean'." ]) @>
```

Comment:
The `$` here is not actually needed since the string is not interpolated. You may or may not care. This is the first of 11 such cases, all in this file.

Reply:
Will fix in the next PR.